### PR TITLE
Add cross-mode re-enroll rejection tests for external auth

### PR DESCRIPTION
**Pull request by CLBRITTON2**
## Summary
- Add four tests proving a same-name identity can't be re-enrolled under a different mode: none to cert, cert to none, cert to token, token to cert
- Extract the full enroll-to-cert and enroll-to-token flows into reusable `completeEnrollToCert` / `completeEnrollToToken` helpers so the new cross-mode tests share one path with the existing completion tests
- Move the completion-flow helpers to the bottom of the file so the suite reads top-down; no behavior change

## External auth tests
- Each follows the same shape: `UpdateExtJwtSigner` to mode A, full enroll under A, `UpdateExtJwtSigner` to mode B, `AddIdentity` with the same name under mode B, then assert rejection and that the original identity file is still valid for mode A

## Helpers
- `completeEnrollToCert` / `completeEnrollToToken` drive a full flow for a name and return the resulting identifier (working signer must already have that mode enabled)
- `createIdentityData` builds the `AddIdentityData` for a given name and `EnrollMode` against the working signer
- `assertIdentityExistsSameName` asserts the rejection: `Success() == false`, `Code` 500, error contains "identity exists with the same name"

### CODE REVIEW

File: tests/integration/external_auth_test.go [hunk 1]
```diff
@@ -315,3 +255,157 @@ func (c *extAuthContext) addEnrollToNoneIdentity(t *testing.T, name string) test
...
+
+		enrollToNone := testutil.ExtJwtSignerSpec{}
+		c.overlay.UpdateExtJwtSigner(t, c.workingSigner.name, enrollToNone)
+		identityEvent := c.addEnrollToNoneIdentity(t, name)
+
+		c.overlay.UpdateExtJwtSigner(t, c.workingSigner.name, testutil.ExtJwtSignerSpec{EnrollToCert: true})
+		enrollToCertIdentity := c.createIdentityData(name, testutil.EnrollModeCert)
+		addResp := testutil.AddIdentity(t, c.zet.Commands, enrollToCertIdentity)
+		c.assertIdentityExistsSameName(t, addResp)
+		testutil.AssertValidUrlEnrolledIdentityFile(t, identityEvent.Id.Identifier, testutil.EnrollModeNone)
```
Comment (dovholuknf) [inline tests/integration/external_auth_test.go:355]: i might have made a "ValidIdentityEnrollToNone" type of func that verifies the identity file doesn't have a key/cert field as well as the url is valid.

File: tests/integration/external_auth_test.go [hunk 2]
```diff
@@ -315,3 +255,157 @@ func (c *extAuthContext) addEnrollToNoneIdentity(t *testing.T, name string) test
...
+		c.overlay.UpdateExtJwtSigner(t, c.workingSigner.name, enrollToNone)
+
+		controllerBase := c.overlay.ControllerHostPort()
+		enrollToNoneIdentity := testutil.AddIdentityData{
+			IdentityFilename: name,
+			ControllerURL:    &controllerBase,
+		}
+		addResp := testutil.AddIdentity(t, c.zet.Commands, enrollToNoneIdentity)
+		c.assertIdentityExistsSameName(t, addResp)
+		testutil.AssertValidUrlEnrolledIdentityFile(t, identifier, testutil.EnrollModeCert)
```
Comment (dovholuknf) [inline tests/integration/external_auth_test.go:377]: also need to verify the key/cert are still here

File: tests/integration/external_auth_test.go [hunk 3]
```diff
@@ -315,3 +255,157 @@ func (c *extAuthContext) addEnrollToNoneIdentity(t *testing.T, name string) test
...
+	})
+}
+
+func (c *extAuthContext) testEnrollToCertThenTokenRejected(t *testing.T) {
+	testutil.RunTestWithTimeout(t, func(t *testing.T) {
+		name := testutil.IdentityName(t)
+		t.Cleanup(func() { _ = c.overlay.PurgeIdentityByExternalId(c.idp.ExternalID) })
```
Comment (dovholuknf) [inline tests/integration/external_auth_test.go]: `c.overlay.PurgeIdentityByExternalId` could just be an empty arg func and all these become:

```
t.Cleanup(c.overlay.PurgeIdentityByExternalId)
```

File: tests/integration/external_auth_test.go [hunk 4]
```diff
@@ -315,3 +255,157 @@ func (c *extAuthContext) addEnrollToNoneIdentity(t *testing.T, name string) test
...
+
+		c.overlay.UpdateExtJwtSigner(t, c.workingSigner.name, testutil.ExtJwtSignerSpec{EnrollToCert: true})
+		identifier := c.completeEnrollToCert(t, name)
+
+		c.overlay.UpdateExtJwtSigner(t, c.workingSigner.name, testutil.ExtJwtSignerSpec{EnrollToToken: true})
+		enrollToTokenIdentity := c.createIdentityData(name, testutil.EnrollModeToken)
+		addResp := testutil.AddIdentity(t, c.zet.Commands, enrollToTokenIdentity)
+		c.assertIdentityExistsSameName(t, addResp)
+		testutil.AssertValidUrlEnrolledIdentityFile(t, identifier, testutil.EnrollModeCert)
```
Comment (dovholuknf) [inline tests/integration/external_auth_test.go:393]: and verify key/cert still there and probably that they didn't change

**Review by dovholuknf — APPROVED** [review]

File: tests/integration/external_auth_test.go (hunk 2 above)
Comment (dovholuknf) [reply tests/integration/external_auth_test.go:377]: nevermind - chris pointed out that it does indeed do this already :)
